### ISSUE TITLE
 deeplab v3+ deeplab_demo.ipynb crash  in windows 10 python 3.6

### ISSUE THREAD

**Issue by LightRayGo** [tensorflow/models#3648]
------------------------

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:windows 10
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:1.6.0
- **CUDA/cuDNN version**:9.0
- **GPU model and memory**:1080

### Describe the problem
I have setup follow https://www.tensorflow.org/get_started/os_setup.html 
with windows and testing the installation https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/installation.md
`python deeplab/model_test.py`  is ok.

```
D:\DevelopEnv\models\research>python deeplab/model_test.py
e:\ProgramData\Anaconda3\lib\site-packages\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
2018-03-19 21:30:28.612390: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\platform\cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2018-03-19 21:30:29.020673: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1212] Found device 0 with properties:
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.873
pciBusID: 0000:01:00.0
totalMemory: 8.00GiB freeMemory: 6.59GiB
2018-03-19 21:30:29.161326: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1212] Found device 1 with properties:
name: GeForce GTX 1080 major: 6 minor: 1 memoryClockRate(GHz): 1.873
pciBusID: 0000:02:00.0
totalMemory: 8.00GiB freeMemory: 6.59GiB
2018-03-19 21:30:29.173972: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1227] Device peer to peer matrix
2018-03-19 21:30:29.177940: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1233] DMA: 0 1
2018-03-19 21:30:29.185130: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1243] 0:   Y N
2018-03-19 21:30:29.188656: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1243] 1:   N Y
2018-03-19 21:30:29.194402: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1312] Adding visible gpu devices: 0, 1
2018-03-19 21:30:30.179120: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 2457 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-03-19 21:30:30.278496: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 2457 MB memory) -> physical GPU (device: 1, name: GeForce GTX 1080, pci bus id: 0000:02:00.0, compute capability: 6.1)
e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\framework\tensor_util.py:560: DeprecationWarning: The binary mode of fromstring is deprecated, as it behaves surprisingly on unicode inputs. Use frombuffer instead
  return np.fromstring(tensor.tensor_content, dtype=dtype).reshape(shape)
deeplab/model_test.py:81: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals(len(scales_to_logits), expected_num_logits[i])
2018-03-19 21:30:33.124090: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1312] Adding visible gpu devices: 0, 1
2018-03-19 21:30:33.132516: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 2457 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-03-19 21:30:33.142920: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 2457 MB memory) -> physical GPU (device: 1, name: GeForce GTX 1080, pci bus id: 0000:02:00.0, compute capability: 6.1)
e:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\util\tf_inspect.py:45: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
  if d.decorator_argspec is not None), _inspect.getargspec(target))
.2018-03-19 21:30:38.066761: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1312] Adding visible gpu devices: 0, 1
2018-03-19 21:30:38.071123: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 2457 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-03-19 21:30:38.081243: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 2457 MB memory) -> physical GPU (device: 1, name: GeForce GTX 1080, pci bus id: 0000:02:00.0, compute capability: 6.1)
....
----------------------------------------------------------------------
Ran 5 tests in 13.553s

OK
```

but when I run deeplab_demo.ipynb in Jupyter Notebook the python crash in
`    resized_im, seg_map = model.run(orignal_im)`
I have modify the code to support python3
`import StringIO  ` ---->
`import io`
All print change to print()
I just test " Run on sample images" not "Run on internet images"
The model was download "deeplab_model.tar.gz"

**Comment by YknZhu** [comment]
Hmm the demo notebook runs a full xception model which might exceed GPU memory. Could you try inference on CPU (setting tf.device('/cpu:*') )?

**Comment by LightRayGo** [comment]
Think you, I have add tf.device('/cpu:*') to __init__  but it still not work,

1.     def __init__(self, tarball_path):
2.         """Creates and loads pretrained deeplab model."""
3.         self.graph = tf.Graph()
4.         #tf.device('/cpu:*') 
5.         graph_def = None
6.         # Extract frozen graph from tar archive.
7.         tar_file = tarfile.open(tarball_path)
8.         for tar_info in tar_file.getmembers():
9.             if _FROZEN_GRAPH_NAME in os.path.basename(tar_info.name):
10.                 file_handle = tar_file.extractfile(tar_info)
11.                 graph_def = tf.GraphDef.FromString(file_handle.read())
12.                 break
13. 
14.         tar_file.close()
15.         
16.         if graph_def is None:
17.             raise RuntimeError('Cannot find inference graph in tar archive.')
18.             print('Cannot find inference graph in tar archive.')
19. 
20.         with self.graph.as_default():      
21.             tf.import_graph_def(graph_def, name='')
22.         with tf.device('/cpu:*'): 
23.            self.sess = tf.Session(graph=self.graph)

But I  change this it is work  Line 16

1. import collections
2. import os
3. import io
4. import sys
5. import tarfile
6. import tempfile
7. import urllib
8. 
9. from IPython import display
10. from ipywidgets import interact
11. from ipywidgets import interactive
12. from matplotlib import gridspec
13. from matplotlib import pyplot as plt
14. import numpy as np
15. from PIL import Image
16. os.environ["CUDA_VISIBLE_DEVICES"]="-1"   
17. import tensorflow as tf
18. 
19. if tf.__version__ < '1.5.0':
20.     raise ImportError('Please upgrade your tensorflow installation to v1.5.0 or newer!')
21. 
22. # Needed to show segmentation colormap labels
23. sys.path.append('utils')
24. import get_dataset_colormap

It is like exceed GPU memory , How much GPU memory the demo need? I have two GTX 1080. How can I change to support GPU ?

**Comment by YknZhu** [comment]
Hmm `os.environ["CUDA_VISIBLE_DEVICES"]="-1"` effectively masks out all GPUs? BTW I believe you need to move ` with tf.device('/cpu:*')` to line 20 `self.graph.as_default():  ` ;)

xception deeplab inference unfortunately does take a lot of GPU memory... By default tensorflow would try to put the full model in one device. Please refer to https://www.tensorflow.org/programmers_guide/using_gpu for using multiple GPUs (might be pretty difficult to do so in the ipython notebook demo).

**Comment by LightRayGo** [comment]
I see MobileNetv2  will provide  in the next update. Hope this is faster and can run in pc with GPU.Think you！

**Comment by YknZhu** [comment]
Haha please stay tuned :)

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
